src/wp-includes/blocks/index.php

```diff
@@ -155,3 +155,20 @@ function register_core_block_types_from_metadata() {
 	}
 }
 add_action( 'init', 'register_core_block_types_from_metadata' );
+
+/**
+ * Registers the core block metadata collection.
+ *
+ * This function is hooked into the 'init' action with a priority of 9,
+ * ensuring that the core block metadata is registered before the regular
+ * block initialization that happens at priority 10.
+ *
+ * @since 6.7.0
+ */
+function wp_register_core_block_metadata_collection() {
+	wp_register_block_metadata_collection(
```

Comment:
So my hope for the future is the following to be possible in Gutenberg:
```php
function gutenberg_register_core_block_metadata_collection() {
	wp_register_block_metadata_collection(
		BLOCKS_PATH,
		GUTENBERG_BLOCKS_PATH . 'blocks-json.php'
	);
}
remove_action( 'init', 'wp_register_core_block_metadata_collection' );
add_action( 'init', 'gutenberg_register_core_block_metadata_collection', 9 );
```

Reply:
Oh right! I think technically that already _should_ be possible once this PR lands? 🤔